### PR TITLE
Tobin gh399 merge michael fix

### PR DESCRIPTION
**Pull request by tobine**
Rebase of Michael's #399 with fence tracking clean-up.

### CODE REVIEW

**Comment by tobine** [comment]
:+1: Michael once you ok that this merge is clean I'll push this to master barring any issues from others.

**Comment by mark-lunarg** [comment]
Tobin, I assume you were going to push both, so I'm gonna close the other PR with the tests in it.  Thanks.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -4976,17 +5015,24 @@ static void trackCommandBuffers(layer_data *my_data, VkQueue queue, uint32_t sub
             }
         }
     }
+}
+
+static void markCommandBuffersInFlight(layer_data *my_data, VkQueue queue, uint32_t submitCount, const VkSubmitInfo *pSubmits,
```
Comment (lentinem) [inline layers/core_validation.cpp]: Looks like clang-format got messed up here.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -4833,6 +4834,36 @@ static bool validateAndIncrementResources(layer_data *my_data, GLOBAL_CB_NODE *p
...
+                                         " with index %d which was guarded by unsignaled event %" PRIu64 ".",
+                                         (uint64_t)(queryEventsPair.first.pool), queryEventsPair.first.index, (uint64_t)(event));
+                }
+            }
+        }
+    }
+    return skip_call;
+}
+// Decrement cmd_buffer in_use and if it goes to 0 remove cmd_buffer from globalInFlightCmdBuffers
+static inline void removeInFlightCmdBuffer(layer_data *dev_data, VkCommandBuffer cmd_buffer, VkQueue queue) {
```
Comment (lentinem) [inline layers/core_validation.cpp]: My bad but this doesn't need queue as an arg anymore.

**Pull request closed without merging**